src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -14,10 +14,128 @@
+   * use by the test framework or the servlet's init() function.
+   */
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
+
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    request.setAttribute("totalUsers", userStore.getUsers().size());
+    request.setAttribute("totalConversations", conversationStore.getAllConversations().size());
+    request.setAttribute("totalMessages", messageStore.getTotalMessages());
+    request.setAttribute("mostActive", getMostActiveUser());
+    request.setAttribute("newestUser", getNewestUser());
+    request.setAttribute("wordiestIU", getWordiestUser());
+    request.getRequestDispatcher("/WEB-INF/view/admin.jsp").forward(request, response);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    String username = (String) request.getSession().getAttribute("user");
+    Boolean admin = userStore.getUser(username).isAdmin();
+    if (!admin) {
+      request.setAttribute("error", "You are not an Administrator.");
+      request.getRequestDispatcher("/WEB-INF/view/index.jsp").forward(request, response);
+    }
+    response.sendRedirect("/admin");
+  }
+
+  /**
+   * Returns the name of most active user. The most active user is defined by the user that sends
+   * the most Messages.
+   */
+  public String getMostActiveUser() {
+    User currentMostActiveUser = userStore.getUsers().get(0);
+    int currentMostMessages = 0;
+    for (User user : userStore.getUsers()) {
+      int messageCount = messageStore.getMessagesByUser(user.getId()).size();
+      if (messageCount > currentMostMessages) {
+        currentMostMessages = messageCount;
+        currentMostActiveUser = user;
+      }
+    }
+    return currentMostActiveUser.getName();
+  }
+
+  /** Returns the name of the most recently added User. */
+  public String getNewestUser() {
+    return userStore.getUsers().get(userStore.getUsers().size() - 1).getName();
+  }
+
+  /**
+   * Returns the name of the wordiest user. The wordiest user is the user that has the highest count
+   * of characters in all the Messages he/she sent, whitespace is excluded.
+   */
+  public String getWordiestUser() {
+    User currentWordiestUser = userStore.getUsers().get(0);
+    int currentWordiest = 0;
+    for (User user : userStore.getUsers()) {
+      int wordCount = 0;
+      List<Message> messageList = messageStore.getMessagesByUser(user.getId());
+      for (Message message : messageList) {
+        message.getContent().replaceAll("\\s+", "");
```

Comment:
`replaceAll` doesn't actually change the string in-place; it returns a modified version of the string, which isn't being saved.  So your removing whitespace here, but the actual count in the next line still includes the whitespace.

This makes me think it's a good time for add a unit test ;-)